ci/deps/azure-37-locale.yaml

```diff
@@ -18,6 +18,9 @@ dependencies:
   - pymysql
   - pytables
   - python-dateutil
+  # XXX We should be testing ``python=3.7*`` here, but `moto` is
```

Comment:
you can just fix this, add moto with a min version to the pip section

Reply:
For what I understood, the moto version compatible with Python 3.7 is not yet released. Did I misunderstood something?

Reply:
@datapythonista @jreback 
Did you see my comment above https://github.com/pandas-dev/pandas/pull/23727#discussion_r234713038 ?
> @datapythonista @TomAugspurger 
Moto is supporting python 3.7. starting with 1.3.7 (the latest version), but that isn't reflected in the requirements yet: spulec/moto#1886
It's possible to install it through pip - I did that in #23731, but then the problem is that 1.3.7 requires a boto version that will run into #23754, probably because of a moto issue: spulec/moto#1941

The credentials issue would be resolved (or rather deferred) by #23731, which makes all `boto` tests use the `s3_resource` fixture, which would turn those `NoCredentialsErrors` into skips.

Reply:
still if you can just pip install it i am not sure of the problem.

Reply:
Moved to pip, let's see if tests pass. For @h-vetinari comment I understood that even pip version would make the tests fail.